src/cloud/components/onboarding/useGetUserStatus.ts

```diff
@@ -0,0 +1,116 @@
+import {useEffect, useState, useCallback} from 'react'
+import {getOrgsUsage} from 'src/client'
+import {fromFlux} from '@influxdata/giraffe'
+import {usageStatsCsv} from 'src/shared/utils/mocks/usageStats.mocks'
+import {useSelector} from 'react-redux'
+import {getOrg} from 'src/organizations/selectors'
+import {event} from 'src/cloud/utils/reporting'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
+import {Table} from '@influxdata/giraffe'
+
+export enum USER_PILOT_USER_STATUS {
+  NEW_USER = 'NEW_USER',
+  DELETED_ALL_DATA = 'DELETED_ALL_DATA',
+  NON_WRITING_USER = 'NON_WRITING_USER',
+  ACTIVE_USER = 'ACTIVE_USER',
+  ACTIVE_WRITING_USER = 'ACTIVE_WRITING_USER',
+  WRITING_NOT_READING_USER = 'WRITING_NOT_READING_USER',
+}
+export const getUserStatus = (table: Table): USER_PILOT_USER_STATUS[] => {
+  const dataStates: USER_PILOT_USER_STATUS[] = []
+
+  const measurement = (table.getColumn('_measurement') as string[]) ?? []
+
+  // this looks up storage amounts to see if the user has stored data
+  const storageBytes = measurement.find((_v, i) => {
+    return (
+      measurement[i] === 'storage_usage_bucket_bytes' &&
+      table.getColumn('_value', 'number')[i] > 0
+    )
+  })
+
+  // This looks up requests that are POSTs to our query endpoints for writing data
+  const requestInByte = measurement.find((_v, i) => {
+    return (
+      measurement[i] === 'http_request' &&
+      ['/api/v2/query', '/query'].includes(
+        table.getColumn('endpoint', 'string')[i]
+      ) &&
+      table.getColumn('_field')[i] === 'req_bytes' &&
+      table.getColumn('_value', 'number')[i] > 0
```

Comment:
is it possible to have other data types here? I don't know if a `long` is possible, but it might be worth checking

Reply:
pretty sure Giraffe handles that stuff, so we should be G2G